[Java][JVM] Move java specific import mappers to JVM specific code gen

agilob opened the pull request:
<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

jimschubert commented on modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:
```diff
@@ -1472,26 +1472,6 @@ public DefaultCodegen() {
         reservedWords = new HashSet<String>();
 
-        // TODO: Move Java specific import mappings out of DefaultCodegen.
-        importMapping = new HashMap<String, String>();
```
I think many of these are used by Kotlin, and maybe also Groovy. I think the TODO should have read `JVM specific`. I think the only Java specific ones are the joda time.

agilob replied:
Updated for java, scala, kotlin, groovy

agilob commented:
CircleCi fails with "new" files:
```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   samples/client/petstore/elixir/.openapi-generator/FILES
	modified:   samples/server/petstore/php-laravel/.openapi-generator/FILES

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	samples/client/petstore/elixir/lib/openapi_petstore/model/file.ex
	samples/client/petstore/elixir/lib/openapi_petstore/model/list.ex
	samples/server/petstore/php-laravel/lib/app/Models/File.php
	samples/server/petstore/php-laravel/lib/app/Models/ModelList.php

no changes added to commit (use "git add" and/or "git commit -a")
```

These arent my changes

agilob commented:
bump @jimschubert @wangzw

jimschubert commented:
Looks good, thanks!

wing328 commented:
```
	samples/client/petstore/elixir/lib/openapi_petstore/model/file.ex
	samples/client/petstore/elixir/lib/openapi_petstore/model/list.ex
	samples/server/petstore/php-laravel/lib/app/Models/File.php
	samples/server/petstore/php-laravel/lib/app/Models/ModelList.php
```

I wonder if these files should be created as a result of this PR.

wing328 commented:
Just did another review and confirmed these are correct output.

Thanks @agilob again for the bug fixes.
